### ISSUE TITLE
destroy/pump

### ISSUE THREAD

**Issue by calvinmetcalf** [nodejs/readable-stream#283]
since pump relies on destroy, we probably want destroy in before we do pump.

**Comment by mcollina** [comment]
I would say so, let's wait for https://github.com/nodejs/node/pull/12925 to land.

**Comment by mcollina** [comment]
Relevant issues:

https://github.com/mafintosh/pump/issues/17
https://github.com/nodejs/node/issues/1834

**Comment by calvinmetcalf** [comment]
I have a pull I've started a bit back that I can clean up with a bit and submit it, I was waiting on destroy to land first

**Comment by mcollina** [comment]
@calvinmetcalf I think you can fire that, and we can discuss it in the issue tracker.

**Comment by calvinmetcalf** [comment]
yeah I need to clean it up, I had been waiting on destroy to land

**Comment by calvinmetcalf** [comment]
nodejs/node#13506

**Comment by SimenB** [comment]
Any news here? Or is it blocked on landing in node core? Would love to be able to just depend on readable stream 🙂

**Comment by SimenB** [comment]
My understanding of stream internals are next to none, so I'm unfortunately not much help with that 🙁

**Comment by prog1dev** [comment]
https://github.com/nodejs/node/pull/12925 landed in Version 8.0.0, 2017-05-30
